tests/integration/integration_util_test.go

```diff
@@ -343,6 +343,49 @@ func (t *nonfatalT) Fatalf(msg string, args ...interface{}) {
 	t.messages = append(t.messages, fmt.Sprintf(msg, args...))
 }
 
+func testConstructMethods(t *testing.T, lang string, dependencies ...string) {
+	t.Parallel()
+
+	testDir := "construct_component_methods"
+	runComponentSetup(t, testDir)
+
+	componentDirs := []string{
+		"testcomponent",
+		"testcomponent-python",
+		"testcomponent-go",
+	}
+	for _, componentDir := range componentDirs {
+		componentDir := componentDir
+		t.Run(componentDir, func(t *testing.T) {
+			localProvider := integration.LocalDependency{
+				Package: "testcomponent", Path: filepath.Join(testDir, componentDir),
+			}
+			integration.ProgramTest(t, &integration.ProgramTestOptions{
+				Dir:            filepath.Join(testDir, lang),
+				Dependencies:   dependencies,
+				LocalProviders: []integration.LocalDependency{localProvider},
+				Quick:          true,
+				ExtraRuntimeValidation: func(t *testing.T, stackInfo integration.RuntimeValidationStackInfo) {
+					assert.Equal(t, "Hello World, Alice!", stackInfo.Outputs["message"])
+
+					// TODO[pulumi/pulumi#12471]: Only the Go SDK has been fixed such that rehydrated
```

Comment:
So you've fixed the tests to output these values, but they're still the wrong values for now?

Reply:
Sorry, I should have been more clear in the PR description. This is a follow-up to https://github.com/pulumi/pulumi/pull/12494.

I fixed part of https://github.com/pulumi/pulumi/issues/12471 in the Go SDK for a customer in https://github.com/pulumi/pulumi/pull/12494. The particular problem the customer cares about shows up in the implementation of a method in a provider written in Go. In https://github.com/pulumi/pulumi/pull/12494, I modified the Go program of the `construct_component_methods` integration test to verify it's getting back the appropriate dependencies from a method implemented in a provider in Go. But I should have also updated the Node.js and Python programs to do the same thing when calling the method implemented in `testcomponent-go`, which is what this PR does.

There are still subsequent fixes that need to be made in the other SDKs for https://github.com/pulumi/pulumi/issues/12471, that will fix the behavior in `testcomponent` and `testcomponent-python`. Once those fixes are in, we can remove this TODO.